### PR TITLE
Added Sections For Zero Trust and Defense In Depth in SSDLC module 

### PR DESCRIPTION
**Pull request by stilwellc**
We can add Create quiz / example question for Sobelow #27 to this section since I mention Sobelow if it fits best here 

We can also make a quiz question for Cloak if needed

### CODE REVIEW

File: modules/3-ssdlc.livemd [hunk 1]
```diff
@@ -75,18 +76,78 @@ If the answer to one or more of those questions is yes, consider putting a limit
...
+* Virtual Private Networks (VPN)
+
+### Physical 
+
+* Biometrics 
+* Data-centric security 
+* Physical Security (such as locked server rooms) 
+
+### Reference
+- https://www.forcepoint.com/cyber-edu/defense-depth
```
Comment (houllette) [inline modules/3-ssdlc.livemd:153]:
```suggestion
1. https://www.forcepoint.com/cyber-edu/defense-depth
```

File: modules/3-ssdlc.livemd [hunk 2]
```diff
@@ -75,18 +76,78 @@ If the answer to one or more of those questions is yes, consider putting a limit
...
 **Entity: generic term for an arbitrary process, user, program, etc. found within a Data System*
 
-### Benefits of the Principle
+#### Benefits of the Principle
 
 * **Better Data System Stability** - When an entity is limited in the scope of changes it can make to a system, it is easier to test its possible actions and interactions in the context of the Data System.
 * **Better Data System Security** - When an entity is limited in the system-wide actions it may perform, vulnerabilities / compromises in one application cannot be used to exploit the rest of the business or adjacent Data Systems.
 * **Ease of Deployment** - In general, the fewer privileges an entity requires, the easier it is to deploy within a larger environment.
   <br /><br />
-  [**<- Previous Module: OWASP**](./2-owasp.livemd) || [**Next Module: GraphQL Security ->**](./4-graphql.livemd)
```
Comment (houllette) [inline modules/3-ssdlc.livemd:92]: This probably shouldn't be deleted, might've just gotten lost during editing :)

Comment (stilwellc) [reply modules/3-ssdlc.livemd:92]: Good catch haha

**Comment by houllette** [comment]
> We can add Create quiz / example question for Sobelow https://github.com/podium/elixir-secure-coding/issues/27 to this section since I mention Sobelow if it fits best here

I'm of the opinion that we probably shouldn't have the Sobelow quiz question here and instead have it in the 8-cici.livemd instead! I also have a related comment about the introduction of those concepts as part of this module, standby for that.

File: modules/3-ssdlc.livemd [hunk 3]
```diff
@@ -9,7 +9,8 @@ Welcome to Part 3! This section is dedicated to discussing some of the more abst
 * [No Secrets In Code](#no-secrets-in-code)
 * [Making Secret Rotation Easy](#making-secret-rotation-easy)
 * [Rate Limiting](#rate-limiting)
-* [Principle of Least Privilege](#principle-of-least-privlege)
+* [Zero Trust Model](#zero-trust-model)
+* [Defense In Depth](#defense-in-depth)
```
Comment (houllette) [inline modules/3-ssdlc.livemd:13]: Great change! We should probably maintain the standard that subsections are sub-bullets in the Table of Contents and link out to their respective sections (see: the [OWASP Module](https://github.com/podium/elixir-secure-coding/blob/main/modules/2-owasp.livemd) for an example).

File: modules/3-ssdlc.livemd [hunk 4]
```diff
@@ -75,18 +76,78 @@ If the answer to one or more of those questions is yes, consider putting a limit
...
+Zero Trust is designed to contain attackers so that they can not move laterally. You may be asking what does that even mean? In network security, “lateral movement” is when an attacker moves within a network after gaining access to it, which can be very difficult to detect. 
+
+Zero Trust helps contain attackers because the access is segmented and has to be reestablished periodically, limiting them from moving across to other microsegments within the network. 
+
+### Multi Factor Authentication (MFA) 
+
+It's no surprise that MFA is a core part of the Zero Trust Model. Systems using MFA require more than one piece of evidence to authenticate a user, with the most common form being a one time password (OTP). 
+
+### Reference
+- https://www.cloudflare.com/learning/security/glossary/what-is-zero-trust/
```
Comment (houllette) [inline modules/3-ssdlc.livemd:118]:
```suggestion
1. https://www.cloudflare.com/learning/security/glossary/what-is-zero-trust/
```

File: modules/3-ssdlc.livemd [hunk 5]
```diff
@@ -75,18 +76,81 @@ If the answer to one or more of those questions is yes, consider putting a limit
...
+
+* Authentication and password security
+  * Hashing passwords 
+  * Multi factor authentication (MFA) 
+* Encryption 
+  * [Cloak](https://github.com/danielberkompas/cloak) is an Elixir encryption library that implements several best practices and conveniences making it easy for developers 
+* Security Tooling 
+  * Vulnerability Scanners
+    * SAST supporting Elixir: [Semgrep](https://semgrep.dev/docs/supported-languages/) or [Sobelow](https://github.com/nccgroup/sobelow)
+  * Dependence Scanners
```
Comment (houllette) [inline modules/3-ssdlc.livemd:134]:
```suggestion
  * Dependency Scanners
```

File: modules/5-elixir.livemd [hunk 6]
```diff
@@ -204,4 +221,45 @@ user_input = "some_string_which_obviously_isnt_the_same_as_the_password"
...
+** We have decided that we do not want this ETS table to be read from other processes, so try making it private:
+
+```elixir
+cool_table = :ets.new(:cool_table, [])
+```
+
+## Reference
+
+1. https://erlef.github.io/security-wg/secure_coding_and_deployment_hardening/sensitive_data.html
+
```
Comment (houllette) [inline modules/5-elixir.livemd:264]:
```suggestion
```

Comment (houllette) [reply modules/5-elixir.livemd:264]: I think that this chunk from your other PR (https://github.com/podium/elixir-secure-coding/pull/51) got pulled into this one, removing it here to not conflict with the changes over in your other PR

File: modules/5-elixir.livemd [hunk 7]
```diff
@@ -13,10 +13,11 @@ But even the dullest blades can hurt someone! This module goes over Elixir speci
 ## Table of Contents
 
 * [Atom Exhaustion](#atom-exhaustion)
-* [Protecting Sensitive Data](#protecting-sensitive-data)
+* [Serialization and Deserialization](#serialization-and-deserialization)
 * [Untrusted Code](#untrusted-code)
 * [Timing Attacks](#timing-attacks)
 * [Boolean Coercion](#boolean-coercion)
+* [Protecting Sensitive Data](#protecting-sensitive-data)
```
Comment (houllette) [inline modules/5-elixir.livemd:20]:
```suggestion
```

File: modules/5-elixir.livemd [hunk 8]
```diff
@@ -61,6 +62,22 @@ IO.puts("Are you protected against Atom Exhaustion?")
...
+## Serialization and Deserialization
+
+### Description
+
+Deserialization of untrusted input can result in atom creation, which can lead to your application being vulnerable to denial of service (DOS) attacks. When you do use a deserialization library, make sure that the library does not create arbitrary atoms: either configure the library to return strings/binaries or enable schema validation to constrain the input
+
+### Prevention
+
+* Use the :safe option when calling :erlang.binary_to_term/2 on untrusted input (should be familiar from atom exhaustion 😀)
+* Prevent function deserialisation from untrusted input, e.g. using Plug.Crypto.non_executable_binary_to_term/1,2
```
Comment (houllette) [inline modules/5-elixir.livemd:74]:
```suggestion
* Use the :safe option when calling `:erlang.binary_to_term/2` on untrusted input (should be familiar from atom exhaustion 😀)
* Prevent function deserialisation from untrusted input, e.g. using `Plug.Crypto.non_executable_binary_to_term/1,2`
```

File: modules/3-ssdlc.livemd [hunk 9]
```diff
@@ -75,18 +81,86 @@ If the answer to one or more of those questions is yes, consider putting a limit
...
-  [**<- Previous Module: OWASP**](./2-owasp.livemd) || [**Next Module: GraphQL Security ->**](./4-graphql.livemd)
+
+### Device Access Control
+
+Zero Trust is not only about controlling user access, but requires strict controls on device access as well. With this, Zero Trust systems need to monitor how many different devices are trying to access their network, ensure that every device is authorized, and assess all devices to make sure they have not been compromised. This further minimizes the attack surface of the network.
+
+### Microsegmentation
+
+Microsegmentation is the practice of breaking up security perimeters into small zones to maintain separate access for separate parts of the network. Some of the benefits of doing so are: 
+* Granular Access Policies- we can create super specific policies for access to each segment!
```
Comment (houllette) [inline modules/3-ssdlc.livemd:108]:
```suggestion
* Granular Access Policies - we can create super specific policies for access to each segment! 
```

File: modules/3-ssdlc.livemd [hunk 10]
```diff
@@ -75,18 +81,86 @@ If the answer to one or more of those questions is yes, consider putting a limit
...
+* Demilitarized zones (DMZ) 
+* Virtual Private Networks (VPN)
+
+### Physical 
+
+* Biometrics 
+* Data-centric security 
+* Physical Security (such as locked server rooms) 
+
+### Reference
```
Comment (houllette) [inline modules/3-ssdlc.livemd:162]:
```suggestion
### Resource
```

File: modules/3-ssdlc.livemd [hunk 11]
```diff
@@ -75,18 +81,86 @@ If the answer to one or more of those questions is yes, consider putting a limit
...
+
+Zero Trust is designed to contain attackers so that they can not move laterally. You may be asking what does that even mean? In network security, “lateral movement” is when an attacker moves within a network after gaining access to it, which can be very difficult to detect. 
+
+Zero Trust helps contain attackers because the access is segmented and has to be reestablished periodically, limiting them from moving across to other microsegments within the network. 
+
+### Multi Factor Authentication
+
+It's no surprise that MFA is a core part of the Zero Trust Model. Systems using MFA require more than one piece of evidence to authenticate a user, with the most common form being a one time password (OTP). 
+
+### Reference
```
Comment (houllette) [inline modules/3-ssdlc.livemd:122]:
```suggestion
### Resource
```

File: modules/5-elixir.livemd [hunk 12]
```diff
@@ -61,6 +62,22 @@ IO.puts("Are you protected against Atom Exhaustion?")
...
+
+### Resources
+
```
Comment (houllette) [inline modules/5-elixir.livemd:77]:
```suggestion
### Resource
```

**Pull request merged**